static/js/lib/search_test.js

```diff
@@ -497,67 +497,206 @@ describe("search functions", () => {
         }
       ]
 
-      const mustQuery = [
-        {
-          term: {
-            object_type: type
-          }
-        },
-        {
-          bool: {
-            should: [
-              {
-                term: {
-                  offered_by: "MITx"
+      assert.deepEqual(buildSearchQuery({ type, text, facets }), {
+        aggs: {
```

Comment:
These expected queries are getting so big that I wonder if maybe they should be read from a separate file, like `/test_json/es_response_01.json` for example.  But that could wait for a future PR.

Reply:
I added a loop for some of the redundant subqueries to make the query shorter